src/analysis/normed_space/star/continuous_functional_calculus.lean

```diff
@@ -0,0 +1,232 @@
+  [has_continuous_star A] [star_module R A] (a : A) [is_star_normal a] :
+  normed_comm_ring (elemental_star_algebra R a) :=
+{ mul_comm := mul_comm, .. subring_class.to_normed_ring (elemental_star_algebra R a) }
+
+instance {R A : Type*} [normed_field R] [star_ring R] [normed_ring A] [normed_algebra R A]
+  [star_ring A] [has_continuous_star A] [star_module R A] (a : A) :
+  normed_algebra R (elemental_star_algebra R a) :=
+star_subalgebra.to_normed_algebra (elemental_star_algebra R a)
+
+instance {R A : Type*} [normed_field R] [star_ring R] [normed_ring A] [normed_algebra R A]
+  [star_ring A] [has_continuous_star A] [star_module R A] (a : A) :
+  normed_space R (elemental_star_algebra R a) :=
+normed_algebra.to_normed_space _
+
+-- without this instance type class search causes timeouts
+noncomputable instance elemental_star_algebra.complex.normed_algebra (a : A) :
+  normed_algebra ℂ (elemental_star_algebra ℂ a) :=
+infer_instance
+
+variables [complete_space A] (a : A) [is_star_normal a] (S : star_subalgebra ℂ A)
+
+/-- This lemma is used in the proof of `star_subalgebra.is_unit_of_is_unit_of_is_star_normal`,
+which in turn is the key to spectral permanence `star_subalgebra.spectrum_eq`, which is itself
+necessary for the continuous functional calculus. Using the continuous functional calculus, this
+lemma can be superseded by one that omits the `is_star_normal` hypothesis. -/
+lemma spectrum_star_mul_self_of_is_star_normal :
+  spectrum ℂ (star a * a) ⊆ set.Icc (0 : ℂ) (∥star a * a∥) :=
+begin
+  -- this instance should be found automatically, but without providing it Lean goes on a wild
+  -- goose chase when trying to apply `spectrum.gelfand_transform_eq`.
+  letI := elemental_star_algebra.complex.normed_algebra a,
+  unfreezingI { rcases subsingleton_or_nontrivial A },
+  { simp only [spectrum.of_subsingleton, set.empty_subset] },
+  { set a' : elemental_star_algebra ℂ a := ⟨a, self_mem ℂ a⟩,
+    refine (spectrum.subset_star_subalgebra (star a' * a')).trans _,
+    rw [←spectrum.gelfand_transform_eq (star a' * a'), continuous_map.spectrum_eq_range],
+    rintro - ⟨φ, rfl⟩,
+    rw [gelfand_transform_apply_apply ℂ _ (star a' * a') φ, map_mul φ, map_star φ],
+    rw [complex.eq_coe_norm_of_nonneg star_mul_self_nonneg, ←map_star, ←map_mul],
+    exact ⟨complex.zero_le_real.2 (norm_nonneg _),
+      complex.real_le_real.2 (alg_hom.norm_apply_le_self φ (star a' * a'))⟩, }
+end
+
+variables {a}
+
+/-- This is the key lemma on the way to establishing spectral permanence for C⋆-algebras, which is
+established in `star_subalgebra.spectrum_eq`. This lemma is superseded by
+`star_subalgebra.coe_is_unit`, which does not require an `is_star_normal` hypothesis and holds for
+any closed star subalgebra. -/
+lemma elemental_star_algebra.is_unit_of_is_unit_of_is_star_normal (h : is_unit a) :
+  is_unit (⟨a, self_mem ℂ a⟩ : elemental_star_algebra ℂ a) :=
+begin
+  nontriviality A,
```

Comment:
could you sketch the proof briefly in a comment at the beginning of the proof?

Reply:
Well, perhaps not *briefly*, but I've now documented this proof wth comments.